api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java

```diff
@@ -270,6 +270,46 @@ public void updateRecentWorkspaces() {
     assertThat(actualIds).containsAll(expectedIds);
   }
 
+  @Test
+  public void updateRecentWorkspaces_multipleUsers() {
+    mockWorkspaces.forEach(
+        workspace -> {
+          // Need a new 'now' each time or else we won't have lastAccessDates that are different
+          // from each other
+          workspaceService.updateRecentWorkspaces(
+              workspace,
+              USER_ID,
+              Timestamp.from(NOW.minusSeconds(mockWorkspaces.size() - workspace.getWorkspaceId())));
+        });
+    long OTHER_USER_ID = 2L;
+    workspaceService.updateRecentWorkspaces(
```

Comment:
To reproduce the bug, shouldn't this happen before the other updates?

Reply:
Argh, yes, you're right.